### PR TITLE
Bug 1128394 - Enable newly installed layout when confirmed

### PR DESCRIPTION
**Pull request by timdream**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9705b0172a207f4935ae5b2bc5956c1a5de16bb7

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fb0ee4d1613b95ff6464d4cba4a85c53039744aa

File: apps/keyboard/js/keyboard/settings.js [hunk 1]
```diff
@@ -136,6 +136,36 @@ SettingsPromiseManager.prototype.setOne = function(key, value) {
   return this.set(obj);
 };
 
+// Update the value in the database.
+// You should use this method instead of get() & set() because we will
+// be using the same lock to make sure value will not be overwritten.
+//
+// The callback should take one arguemnt (the current value) and returns
```
Comment (RudyLu) [inline apps/keyboard/js/keyboard/settings.js]: typo: take one `argument`.

File: apps/keyboard/js/settings/layout_item_list.js [hunk 2]
```diff
@@ -86,12 +86,62 @@ DownloadPreference.prototype._isUsingDataConnection = function() {
   return mobileDataConnected;
 };
 
+// This reimplments a subset of KeyboardHelper so that we don't need to
```
Comment (RudyLu) [inline apps/keyboard/js/settings/layout_item_list.js]: typo: This `reimplements`.

File: apps/keyboard/js/settings/layout_item_list_view.js [hunk 3]
```diff
@@ -60,13 +60,26 @@ var Deferred = function() {
...
+// This is a wrapper that wraps <gaia-confirm> into our view management.
+// The reason we did this is because we need to defer the toast from showing,
+// if our containing view is not visible.
+// We also want to queue the showDialog requests.
 var ConfirmationDialogBaseView = function() {
   BaseView.apply(this);
 
-  this.deferred = null;
-};
+  this._isVisible = false;
```
Comment (RudyLu) [inline apps/keyboard/js/settings/layout_item_list_view.js]: I took some time to understand the purpose of these variables, maybe we could add some comments here to describe them.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=20736694222763d47c4b647004ecd957f912fba8

**Pull request merged**